clang/include/clang/CIR/Dialect/Builder/CIRBaseBuilder.h

```diff
@@ -32,6 +32,16 @@ class CIRBaseBuilderTy : public mlir::OpBuilder {
     return create<cir::ConstantOp>(loc, attr.getType(), attr);
   }
 
+  cir::ConstantOp getConstantInt(mlir::Location loc, mlir::Type ty,
+                                 int64_t value) {
+    return create<cir::ConstantOp>(loc, ty, cir::IntAttr::get(ty, value));
+  }
+
+  // Creates constant null value for integral type ty.
+  cir::ConstantOp getNullValue(mlir::Type ty, mlir::Location loc) {
+    return create<cir::ConstantOp>(loc, ty, getZeroInitAttr(ty));
```

Comment:
Both `getConstantInt` and `getNullValue` should be implemented in terms of the `getConstant` above, and diminish the amount of direct calls to `create`.